src/NHibernate/Linq/GroupJoin/NonAggregatingGroupJoinRewriter.cs

```diff
@@ -156,27 +168,34 @@ internal class QuerySourceUsageLocator : RelinqExpressionVisitor
 	{
 		private readonly IQuerySource _querySource;
 		private bool _references;
-		private readonly List<IBodyClause> _clauses = new List<IBodyClause>();
+		private readonly List<IQuerySource> _usages = new List<IQuerySource>();
 
 		public QuerySourceUsageLocator(IQuerySource querySource)
 		{
 			_querySource = querySource;
 		}
 
-		public IList<IBodyClause> Clauses
+		internal bool LeftJoin { get; private set; }
+
+		public IList<IQuerySource> Usages
 		{
-			get { return _clauses.AsReadOnly(); }
+			get { return _usages.AsReadOnly(); }
```

Comment:
What's the point of wrapping collection as readonly on each property access in internal class?

Reply:
Dropped.